README.md

```diff
@@ -100,6 +108,9 @@ Options:
   -t, --test                 Whether to run payload locally. Default: false
       --minify               Whether to minify the payloads where applicable 
                                (experimental). Default: false
+      --ust, --usesimpletype This is to remove additional info only when 
```

Comment:
I dont like having to propagate minify and simpletype for all generators. Why not making it part of minify? Are there any cases where you want to minify, but not use simple types?

Seems like we are adding a lot of code and complexity to enable simple type names for a couple of plugins. I think this should be integrated with minify

Reply:
Minify only removes bit that are not needed.  simple types comes from https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.formatters.formatterassemblystyle?redirectedfrom=MSDN&view=netframework-4.8 
Json.Net for example by default uses simple types but that can be changed in settings. If a vulnerable app has custom setting to use full assembly names, enforcing simple when minifying can make the payload useless. Not using simple when it is supporting it can also make the payloads longer which is vital in certain scenarios. 

This is just a start to have the option so it is not just a few plugins. It requires time to invest on this to see which methods can support simple types and how can that change so I thought we should start from somewhere by adding the option. 

Instead of combining this with minify for the command line, we can combine it with minify when passing it to the generator to make that code simpler - I think I need to restructure how the generator works - I like to pass it an input object rather than passing everything as separate parameters - this way we can easily extend it. So instead of having this:
```
public override object Generate(string cmd, string formatter, Boolean test, Boolean minify, Boolean useSimpleType)
``` 
it will be more like this
```
public override object Generate(InputArgs options)
```

and we can then easily change the InputArgs class to add more options to it to support even more capabilities without changing all the gadgets every time.

What do you think? I will need time to do this but it sounds good to me to do it this way.

Reply:
Ok, that makes sense, and I like the idea of `InputArgs` but I would keep `cmd` and maybe `formatter` as independent args.

Reply:
ok, I will do that. Any reason in particular why cmd shouldn't be part of the inputArgs? I can parse the cmd already when storing it in InputArgs to have it as command+its argument and this will make the code less messy when it comes for me to call the command parser in all the gadgets.

Reply:
I forgot to press unresolved, as I need to see whether I can add cmd to inputArgs too :)

Reply:
No strong opinion on that, so its ok if you want to add cmd as inputArgs